cisco_aci/datadog_checks/cisco_aci/fabric.py

```diff
@@ -209,3 +224,45 @@ def get_fabric_type(self, obj_type):
             return 'pod'
         if obj_type == 'l1PhysIf':
             return 'port'
+
+    def submit_node_metadata(self, node_attrs, tags):
+        vendor = 'cisco_aci'
+        namespace = 'default'
+        node = Node(attributes=node_attrs)
+        id_tags = ['namespace:{}'.format(namespace), 'system_ip:{}'.format(node.attributes.address)]
+        device_tags = [
+            'device_vendor:{}'.format(vendor),
+            'device_namespace:{}'.format(namespace),
+            'device_hostname:{}'.format(node.attributes.dn),
+            'hostname:{}'.format(node.attributes.dn),
+            'system_ip:{}'.format(node.attributes.address),
+            'device_ip:{}'.format(node.attributes.address),
+            'device_id:{}:{}'.format(namespace, node.attributes.address),
+        ]
+        device = DeviceMetadata(
+            device_id='{}:{}'.format(namespace, node.attributes.address),
+            id_tags=id_tags,
+            tags=device_tags + tags,
+            name=node.attributes.dn,
+            ip_address=node.attributes.address,
+            model=node.attributes.model,
+            ad_st=node.attributes.ad_st,
+            vendor=vendor,
+            version=node.attributes.version,
+            serial_number=node.attributes.serial,
+        )
+        self.ndm_metadata(json.dumps(device.model_dump()))
+
+    def submit_interface_metadata(self, eth_attr, address, tags):
+        eth = Eth(attributes=eth_attr)
+        namespace = 'default'
+        interface = InterfaceMetadata(
+            device_id='{}:{}'.format(namespace, address),
+            id_tags=tags,
+            index=eth.attributes.id,
+            name=eth.attributes.name,
+            description=eth.attributes.desc,
+            mac_address=eth.attributes.router_mac,
+            admin_status=eth.attributes.admin_st,
+        )
+        self.ndm_metadata(json.dumps(interface.model_dump(exclude_none=True)))
```

Comment:
Not sure how it works from Python when sending EvP events. In the snmp go code we batch events together to try to group interfaces in the same payload as the device information.
This may not be a high priority but I'm wondering if you looked into it and if you think it's ok to send payloads like this.

Reply:
Also I'm surprised that it works, have you tried it?
The payload our intake expects looks like this:

```go
type NetworkDevicesMetadata struct {
	Subnet           string                 `json:"subnet,omitempty"`
	Namespace        string                 `json:"namespace"`
	Devices          []DeviceMetadata       `json:"devices,omitempty"`
	Interfaces       []InterfaceMetadata    `json:"interfaces,omitempty"`
	IPAddresses      []IPAddressMetadata    `json:"ip_addresses,omitempty"`
	Links            []TopologyLinkMetadata `json:"links,omitempty"`
	NetflowExporters []NetflowExporter      `json:"netflow_exporters,omitempty"`
	Diagnoses        []DiagnosisMetadata    `json:"diagnoses,omitempty"`
	DeviceOIDs       []DeviceOID            `json:"device_oids,omitempty"`
	CollectTimestamp int64                  `json:"collect_timestamp"`
}
```